### PR TITLE
Implement JSDOC

### PR DESCRIPTION
**Pull request by nhtruong**
### Description
This change added jsdoc and a workflow to generate and publish documentation to gh_pages. It also includes documentation for Transport#request and api/bulk. More documentation will also be added incrementally in the subsequent PRs.

### Issues Resolved
closes https://github.com/opensearch-project/opensearch-js/issues/335

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by harshavamsi** [comment]
@nhtruong is there a link to the docs on your fork?

**Comment by nhtruong** [comment]
@harshavamsi Yes here it is: https://nhtruong.github.io/opensearch-js/2.1/
This was created by the workflow included in this PR.
Note that it looks very bare-bone right now with only 2 articles for Transport and Bulk API. And we will need to come up with some text for the homepage in another PR.

**Comment by nhtruong** [comment]
Added docs for all CAT APIs
README.md is used as the homepage for now (It's not great but it's a start)

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,9 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 ### Added
 - Add release details to releasing.md ([319](https://github.com/opensearch-project/opensearch-js/pull/319))
 - Allow overriding the aws service identifier in AwsSigv4Signer ([333](https://github.com/opensearch-project/opensearch-js/pull/333))
+- jsdoc for documentation generation ([#335](https://github.com/opensearch-project/opensearch-js/issues/335))
```
Comment (dblock) [inline CHANGELOG.md:8]: Added jsdoc ...

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,9 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
...
+- Documentation for Transport#request ([#335](https://github.com/opensearch-project/opensearch-js/issues/335))
```
Comment (dblock) [inline CHANGELOG.md:9]: Documented ...

File: api/api/cat.js [hunk 3]
```diff
@@ -92,16 +95,34 @@ function CatApi(transport, ConfigurationError) {
   this[kConfigurationError] = ConfigurationError;
 }
 
+/**
+ * The CAT aliases operation lists the mapping of aliases to indices, plus routing and filtering information.
```
Comment (dblock) [inline api/api/cat.js:99]: Is it `CAT` or `cat`?

**Review by dblock — COMMENTED** [review]
I like it but I'd like to nitpick a little.

File: api/api/cat.js (hunk 3 above)
Comment (nhtruong) [reply api/api/cat.js:99]: CAT is short for "Compact and Aligned Text". Opensearch docs also refers to it as CAT API. So, I think CAT is more accurate.

Comment (dblock) [reply api/api/cat.js:99]: TIL

**Comment by nhtruong** [comment]
@harshavamsi @dblock This PR is now ready for review. I've added documentation for all API methods.
Check out the page on my fork to see what the docs look like: https://nhtruong.github.io/opensearch-js/2.1/

There will be a follow up PR to add this docs to the README after the github pages have been properly setup and tested on this repo.

**Comment by ananzh** [comment]
@nhtruong like these documents!! I just opened the attached url link and saw that the logo is not rendered properly. are you aware of this?

Screen Shot 2022-12-16 at 12 59 47

**Review by dblock — APPROVED** [review]

**Pull request merged**